### ISSUE TITLE
Rancher CLI on Windows partially not worked

### ISSUE THREAD

**Issue by shandyDev** [rancher/rancher#7262]
**Rancher Version:**
v1.2.2
**Docker Version:**
1.12.5
**OS and where are the hosts located? (cloud, bare metal, etc):**
VPS
**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node
**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle
**Steps to Reproduce:**
rancher exec --debug -it 1s218 bash
**Results:** 
FATA[0001] listen unix C:\Users\Shandy\AppData\Local\Temp\docker-sock234811611: socket: An address incompatible with the requested protocol was used.
**Expected:**
Exec in container on remote host.

Hi,
Windows 10 Pro, docker for windows (hyper-v), native docker client, latest rancher cli (0.4.1)
Exec command as example, all rancher docker commands have same behavior.
I tried on both windows shell: git bash & powershell. 

It seems problem is here: 
- https://github.com/rancher/cli/blob/master/cmd/docker.go#L113
- https://github.com/rancher/rancher-docker-api-proxy/blob/master/server.go#L43

**Comment by shandyDev** [comment]
Sorry found dublicate issue https://github.com/rancher/rancher/issues/6949 But I provided more details )

**Comment by jtcressy** [comment]
To add some additional info, this is due to windows currently being fundamentally incompatible with unix sockets. Until someone comes up with a translation layer for winsock<->unix, we wont see these features working. 

Also, before anyone says "But docker for windows works just fine", note that the windows version of docker is a bit of a hack and is just used to talk to a real docker daemon in a vbox/hyperv virtual machine.

**Comment by juliangamble** [comment]
+1

**Comment by shandyDev** [comment]
So, with last windows 10 update (creators update) I found way to work "rancher exec" and so on:
- update docker to latest (I used 17.03)
- install latest WSL (Bash on Ubuntu) and go to bash
- download Rancher CLI for linux x64
- Set up Racnher CLI config & run rancher exec
- profit!!!

On Windows binary is same fail result:
> listen unix C:\Users\Shandy\AppData\Local\Temp\docker-sock746830959: socket: An address incompatible with the requested protocol was used.

I think it work on "Bash on Ubuntu" because docker server now can access via TCP: "docker -H tcp://localhost:2375 info" in both console (win + ubuntu).

So instead of "translation layer for winsock<->unix" I propose change logic connect to docker server in Racher CLI for win platform, it can be just fallback after trying connect via socket, try connect via TCP.

**Comment by metabrain** [comment]
I posted a workaround in the ticket linked by @shandyDev https://github.com/rancher/rancher/issues/6949#issuecomment-323134373  

TL;DR: Use the rancher/cli Docker container to run commands for you in Windows:
`
docker run -e RANCHER_URL=<URL> -e RANCHER_ACCESS_KEY=<KEY> -e RANCHER_SECRET_KEY=<SECRET> -it rancher/cli:latest <CMD>
`

**Comment by deniseschannon** [comment]
Available in v1.6.15-rc2 using v0.6.8-rc2

**Comment by kingsd041** [comment]
Tested with rancher/server:v1.6.15-rc3 with rancher CLI v0.6.8-rc2.
In the rancher/server:v1.6.15-rc3, Rancher CLI can work on windows.
